test/e2e/features_test.go

```diff
@@ -0,0 +1,334 @@
+// much longer to start, but once they do Secret updates show up more quickly, enough for the test to complete in time.
+const webhookKINDConfig = `kind: Cluster
+apiVersion: kind.x-k8s.io/v1alpha4
+nodes:
+- role: control-plane
+  kubeadmConfigPatches:
+  - |
+    kind: KubeletConfiguration
+    configMapAndSecretChangeDetectionStrategy: Get
+    syncFrequency: 3s
+`
+
+// TestWebhookUpdate checks that the webhook updates the certificate indicated by --admission-webhook-cert-file when
+// the mounted Secret updates. This requires E2E because we can't mount Secrets with the locally-run integration
+// test controller instance.
+func TestWebhookUpdate(t *testing.T) {
+	t.Log("configuring all-in-one-dbless.yaml manifest test")
+	t.Parallel()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	t.Log("building test cluster and environment")
+	configFile, err := os.CreateTemp(os.TempDir(), "webhook-kind-config-")
+	require.NoError(t, err)
+	defer os.Remove(configFile.Name())
+	defer configFile.Close()
+	written, err := configFile.Write([]byte(webhookKINDConfig))
+	require.NoError(t, err)
+	require.Equal(t, len(webhookKINDConfig), written)
+
+	clusterBuilder := kind.NewBuilder()
+	clusterBuilder.WithConfig(configFile.Name())
+	if clusterVersionStr != "" {
+		clusterVersion, err := semver.ParseTolerant(clusterVersionStr)
+		require.NoError(t, err)
+		clusterBuilder.WithClusterVersion(clusterVersion)
+	}
+	cluster, err := clusterBuilder.Build(ctx)
+	require.NoError(t, err)
+	addons := []clusters.Addon{}
+	addons = append(addons, metallb.New())
+	if b, err := loadimage.NewBuilder().WithImage(imageLoad); err == nil {
+		addons = append(addons, b.Build())
+	}
+	builder := environments.NewBuilder().WithExistingCluster(cluster).WithAddons(addons...)
+	env, err := builder.Build(ctx)
+	require.NoError(t, err)
+	defer func() {
+		assert.NoError(t, env.Cleanup(ctx))
+	}()
+
+	t.Log("deploying kong components")
+	manifest, err := getTestManifest(t, dblessPath)
+	require.NoError(t, err)
+	deployment := deployKong(ctx, t, env, manifest)
+
+	firstCertificate := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "admission-cert",
+		},
+		Type: corev1.SecretTypeTLS,
+		Data: map[string][]byte{
+			"tls.crt": []byte(tlsPairs[0].Cert),
+			"tls.key": []byte(tlsPairs[0].Key),
+		},
+	}
+
+	secondCertificate := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "admission-cert",
+		},
+		Type: corev1.SecretTypeTLS,
+		Data: map[string][]byte{
+			"tls.crt": []byte(tlsPairs[1].Cert),
+			"tls.key": []byte(tlsPairs[1].Key),
+		},
+	}
+
+	_, err = env.Cluster().Client().CoreV1().Secrets("kong").Create(ctx, firstCertificate, metav1.CreateOptions{})
+	assert.NoError(t, err)
+
+	t.Log("exposing admission service to the test environment")
+	admission, err := env.Cluster().Client().CoreV1().Services("kong").Get(ctx, "kong-validation-webhook",
+		metav1.GetOptions{})
+	assert.NoError(t, err)
```

Comment:
if we can't GET the service, then there's no point in move on from here to try and update it:

```suggestion
	require.NoError(t, err)
```